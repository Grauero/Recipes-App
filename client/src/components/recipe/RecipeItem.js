import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import posed from 'react-pose';

const AnimatedListItem = posed.li({
  shown: { opacity: 1 },
  hidden: { opacity: 0 }
});

const RecipeItem = ({
  _id, imageUrl, name, category
}) => (
  <AnimatedListItem
    className="card"
    style={{ background: `url(${imageUrl}) no-repeat center center / cover` }}
  >
    <span className={category}>{category}</span>
    <div className="card-text">
      <Link to={`/recipes/${_id}`}>
        <h4>{name}</h4>
      </Link>
    </div>
  </AnimatedListItem>
);

RecipeItem.propTypes = {
  _id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired
};

export default RecipeItem;
