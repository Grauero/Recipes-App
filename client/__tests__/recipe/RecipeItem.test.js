import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import RecipeItem from '../../src/components/recipe/RecipeItem';

it('matches snapshot', () => {
  const props = {
    _id: '_id',
    imageUrl: 'imageUrl',
    name: 'name',
    category: 'category'
  };
  const component = shallow(<RecipeItem {...props} />);

  expect(toJSON(component)).toMatchSnapshot();
});
