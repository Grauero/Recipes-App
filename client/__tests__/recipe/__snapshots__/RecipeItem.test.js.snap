// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ForwardRef
  className="card"
  style={
    Object {
      "background": "url(imageUrl) no-repeat center center / cover",
    }
  }
>
  <span
    className="category"
  >
    category
  </span>
  <div
    className="card-text"
  >
    <Link
      replace={false}
      to="/recipes/_id"
    >
      <h4>
        name
      </h4>
    </Link>
  </div>
</ForwardRef>
`;
